### PR TITLE
feat(provider): instrument ScraperExecutor refresh path with AIMD (#2055)

### PR DESCRIPTION
**Pull request by sydlexius**
## What

Instruments the **production scraper refresh path** -- `scraper.Executor.ScrapeAll` (via `getProviderResult`) -- with AIMD rate-limit signals. The executor now records per-provider success / rate-limit feedback into the **same** `AIMDController` instance the orchestrator uses, so both code paths share per-provider rate-limit state instead of diverging.

Previously, when a `ScraperExecutor` was configured, `Orchestrator.FetchMetadata` short-circuited to `executor.ScrapeAll` **before** any AIMD hooks ran. Per `reference_dual_metadata_path.md` the executor is the live **refresh** path in production, so the primary refresh path was uninstrumented: 429 / Retry-After signals during a refresh never fed AIMD.

## Why

Follow-up to **#1747** (AIMD controller, merged as PR #2056). #1747 instrumented the orchestrator search / identify / image / direct-metadata paths but explicitly left the executor refresh path as a tracked gap (a code comment marked it). This PR closes that gap so adaptive back-off / ramp-up applies to the production refresh path too.

## Key changes

- **`NewExecutor` now takes a `*provider.AIMDController`** (nil-safe). When nil, AIMD signals are skipped and the executor behaves exactly as before -- backward compatible.
- **`getProviderResult` emits exactly one composite AIMD signal per invocation**, mirroring the orchestrator's discipline: it accumulates across `GetArtist` + `GetImages`, records the first rate-limit / provider-unavailable error, and at the end fires either `RecordFailure` (a rate-limit error was seen) or `RecordSuccess` (a useful result with no rate-limit error). Ordinary errors (`ErrNotFound`, auth, JSON parse, context cancel) are **not** AIMD signals.
- **Exported `IsRateLimitError` / `RetryAfterDuration`** from `internal/provider` (with package-internal aliases retained for the orchestrator's own `getProviderResult`), so the scraper package can reuse the same rate-limit-error classification.
- **New `GetCurrentLimit`** on `AIMDController` -- lazily initializes state, primarily for tests asserting AIMD signal effects without reaching into unexported `aimdState`.
- **`main.go` wires the shared controller** -- `wireProviders` passes the same `aimdCtrl` to `NewExecutor` that the orchestrator already uses.
- Updated the `FetchMetadata` short-circuit comment to reflect that the executor now records AIMD signals internally.

## Test plan

- `go build ./... && go vet ./...` -- clean.
- `go test -race ./internal/provider/ ./internal/scraper/` -- clean, 0 data races.
- Patch coverage **90.38%** (> gate).
- New tests pin the **exactly-one-signal invariant** via exact `GetCurrentLimit` values, plus coverage for `GetCurrentLimit`, the exported rate-limit helpers (`IsRateLimitError` / `RetryAfterDuration`), and the executor images / not-found / composite paths.

Closes #2055

## Summary by CodeRabbit

* **New Features**
  * Integrated adaptive rate-limiting into the scraper, enabling intelligent adjustment of request rates based on provider feedback and automatic backoff on rate-limit responses.

* **Tests**
  * Added comprehensive test coverage for rate-limit detection, backoff behavior, and edge cases across multiple scenarios.

### CODE REVIEW

File: internal/provider/aimd_test.go [hunk 1]
```diff
@@ -318,6 +319,83 @@ func TestAIMDConcurrencyTwoKeys(t *testing.T) {
...
+	}
+
+	// Fast path: the slot now exists, so a second call reads it under the RLock
+	// and returns the same value.
+	if got := ctrl.GetCurrentLimit(name); got != floor {
+		t.Fatalf("initialized GetCurrentLimit (RLock fast path): expected floor %v, got %v", floor, got)
+	}
+
+	// One additive increase: aimdSuccessThreshold successes raise the limit by
+	// aimdIncrement. The fast path must report the elevated value.
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/provider/aimd_test.go:346]: @codoki False positive — `for range aimdSuccessThreshold` is valid Go 1.22+ range-over-integer syntax (go.dev/doc/go1.22). This module declares `go 1.26.4`. CI confirms it compiles and passes: the providers test shard is GREEN at this HEAD. No change needed.

File: internal/scraper/executor_test.go [hunk 2]
```diff
@@ -1498,3 +1499,351 @@ func TestEffectiveFieldOrdering_NonEmpty_Unchanged(t *testing.T) {
...
+		name: provider.NameAudioDB,
+		getArtFn: func(_ context.Context, _ string) (*provider.ArtistMetadata, error) {
+			if failMode {
+				return nil, &provider.ErrProviderUnavailable{
+					Provider:   provider.NameAudioDB,
+					RetryAfter: 5 * time.Second,
+				}
+			}
+			return &provider.ArtistMetadata{
+				Biography: "A long enough biography for the junk filter, definitely more than fifty characters long.",
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/scraper/executor_test.go:1570]: @codoki False positive — `for i := range primeRounds` is valid Go 1.22+ range-over-integer syntax (go.dev/doc/go1.22). This module declares `go 1.26.4`. CI confirms it compiles and passes: the scan test shard is GREEN at this HEAD. No change needed.
